### ISSUE TITLE
dials.export format=mmcif fails when mmcif.hklout is set by user

### ISSUE THREAD

**Issue by ViktorEGBengtsson** [dials/dials#1204]
Dear all,
Using Dials 2.1.1 I tried exporting to mmcif format during testing of some other code, setting the filename with mmcif.hklout. The result was similar to the following:

```
DIALS 2.1.1-gcdd629a7b-release
The following parameters have been modified:

format = mtz sadabs nxs *mmcif mosflm xds xds_ascii json
mmcif {
  hklout = "integrated.cif"
}
input {
  experiments = integrated.expt
  reflections = integrated.refl
}

Data appears to be unscaled, setting intensity = profile+sum
Read 3185 predicted reflections
Selected 3185 reflections integrated by profile and summation methods
Traceback (most recent call last):
  File "/mnt/e/Linux_programs/dials-v2-1-1/build/../modules/dials/command_line/export.py", line 477, in <module>
    exporter(params, experiments, reflections)
  File "/mnt/e/Linux_programs/dials-v2-1-1/build/../modules/dials/command_line/export.py", line 319, in export_mmcif
    outfile.write(experiments, reflections[0])
  File "/mnt/e/Linux_programs/dials-v2-1-1/modules/dials/util/export_mmcif.py", line 306, in write
    with open(filename, "w") as fh:
UnboundLocalError: local variable 'filename' referenced before assignment
```

The output inserted above was created by running directly in command line.

When I do not set the filename I instead get the following output:

```
DIALS 2.1.1-gcdd629a7b-release
The following parameters have been modified:

format = mtz sadabs nxs *mmcif mosflm xds xds_ascii json
input {
  experiments = integrated.expt
  reflections = integrated.refl
}

Data appears to be unscaled, setting intensity = profile+sum
Data appears to be unscaled, setting mmcif.hklout = 'integrated.cif'
Read 3185 predicted reflections
Selected 3185 reflections integrated by profile and summation methods
Wrote reflections to integrated.cif
```

Personally I will simply skip these tests for now, but it should probably be fixed.

Kind regards,
Viktor

**Comment by Anthchirp** [comment]
Yes, there is a code path here where the variable remains uninitialized. I suspect the fix would be
```diff
diff --git a/util/export_mmcif.py b/util/export_mmcif.py
index ca86a14e9..7aa52d08b 100644
--- a/util/export_mmcif.py
+++ b/util/export_mmcif.py
@@ -43,13 +43,15 @@ def write(self, experiments, reflections):
             ):
                 filename = "scaled.cif"
                 logger.info(
-                    "Data appears to be scaled, setting mmcif.hklout = 'scaled_unmerged.cif'"
+                    "Data appears to be scaled, setting mmcif.hklout = 'scaled.cif'"
                 )
             else:
                 filename = "integrated.cif"
                 logger.info(
                     "Data appears to be unscaled, setting mmcif.hklout = 'integrated.cif'"
                 )
+        else:
+            filename = self.params.mmcif.hklout
 
         # Select reflections
         selection = reflections.get_flags(reflections.flags.integrated, all=True)
```

**Comment by Anthchirp** [comment]
fix backported for the upcoming 2.2.1 release

There won't be another update for DIALS 2.1, but you can edit the `export_mmcif.py` file in your installation locally. The key change is adding the two lines (`else:` ...) shown above.
